File: server/java/user/src/main/java/xyz/hoper/user/service/impl/UserServiceImpl.java
package xyz.hoper.user.service.impl;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import xyz.hoper.user.dao.UserRepository;
import xyz.hoper.user.service.UserService;
import xyz.hoper.user.entity.User;


@Component
class UserServiceImpl implements UserService {

    @Autowired
    private UserRepository userRepository ;


    public User info(Long id ) {
        try {
            return userRepository.findById(id).get();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: server/kotlin/user/src/main/kotlin/xyz/hoper/user/service/impl/UserServiceImpl.kt
package xyz.hoper.user.service.impl


import org.springframework.stereotype.Component
import xyz.hoper.user.service.UserService
import xyz.hoper.user.entity.User


@Component
class UserServiceImpl : UserService {
    override fun info(id: Long): User? {
        try {
            val user = User()
            user.id = id
            user.name = "测试"
            return user
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }
}
